server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -102,6 +105,8 @@ public SegmentLoaderLocalCacheManager(
       );
     }
     locations.sort(COMPARATOR);
```

Comment:
looks like we are already trying to sort via the available free size, 
The issue seems like the order is not updated after a segment is loaded. 
what do you think about sorting the locations after a segment has been loaded ? 
I think that would probably fix the issue in https://github.com/apache/incubator-druid/issues/7641 ?

Reply:
@nishantmonu51 , This probably makes sense. However, one case is when the segment cache location max sizes are skewed (one or few locations with way more availability than others). The sort strategy resorts to selecting the same location again and again until it's availability falls short of others. This will end up having more or less the same behaviour reported in #7641. Round-robin on the other hand will try to distribute the segments across multiple locations there by improving I/O if the locations are backed by different physical drives. Let me know your thoughts.

@dclim and others, let me know your thoughts.